django/contrib/gis/gdal/raster/source.py

```diff
@@ -217,7 +217,7 @@ def _flush(self):
 
     @property
     def vsi_buffer(self):
-        if not self.is_vsi_based:
+        if not self.is_vsi_based or not self.name.startswith(VSI_MEM_FILESYSTEM_BASE_PATH):
```

Comment:
```suggestion
        if not (self.is_vsi_based and self.name.startswith(VSI_MEM_FILESYSTEM_BASE_PATH)):
```
This change should be covered by tests.